inc/saiport.h

```diff
@@ -2354,6 +2416,57 @@ typedef enum _sai_port_attr_t
      */
     SAI_PORT_ATTR_PATH_TRACING_TIMESTAMP_TYPE,
 
+    /**
+     * @brief Read ethernet copper cable pair status.
+     *
+     * Returns pair states sequentially from list index 0 to n (n = number of pairs - 1)
+     * value of n depends on number of pairs in twisted ethernet copper cable
+     * Cable diagnostics triggers ECD once cable pair states is requested, it is synchronous call.
+     *
+     * @type sai_s32_list_t sai_port_cable_pair_state_t
+     * @flags READ_ONLY
+     */
+    SAI_PORT_ATTR_CABLE_PAIR_STATE,
+
+    /**
+     * @brief Get ethernet copper cable pair length (unit metre)
+     *
+     * Returns ethernet cable pair length sequentially from list index 0 to n (n = number of pairs - 1)
+     * value of n depends on number of pairs in twisted ethernet copper cable
+     * Cable diagnostics runs once cable pair length is requested, it is synchronous call.
+     *
+     * @type sai_s32_list_t
+     * @flags READ_ONLY
+     */
+    SAI_PORT_ATTR_CABLE_PAIR_LENGTH,
+
+    /**
+     * @brief Configure ethernet copper cable type to check the cable status
+     *
+     * @type sai_port_cable_type_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PORT_CABLE_TYPE_UNKNOWN
+     * @validonly SAI_PORT_ATTR_MEDIA_TYPE == SAI_PORT_MEDIA_TYPE_COPPER
+     */
+    SAI_PORT_ATTR_CABLE_TYPE,
+
+    /**
+     * @brief Port's PCS DATA PATH Enable
+     *
+     * true: If NO FEC is selected and PCS data path is enabled,
+     * the packet is checked for lane alignment marker and it aligns recovered data in receive direction.
+     * PCS RX link status is set if the packet is properly aligned.
+     * SAI_PORT_DATAPATH_ENABLE_PASS_THROUGH_MODE: Default pass-through data path is used where packet is passed
+     * without any processing.
+     * This attribute is valid only if SAI_PORT_ATTR_FEC_MODE_EXTENDED is selected as SAI_PORT_FEC_MODE_EXTENDED_NONE.
+     *
+     * @type sai_port_datapath_enable_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PORT_DATAPATH_ENABLE_PASS_THROUGH_MODE
+     * @validonly SAI_PORT_ATTR_USE_EXTENDED_FEC == true
```

Comment:
can do valid only on both attribute you listed above?

Reply:
@lguohan valid only meta checks takes only one condition, this is the reason I mentioned other condition above.